subprojects/packagefiles/sqlitecpp/meson.build

```diff
@@ -0,0 +1,298 @@
+    sqlitecpp_args += [
+        '-Wextra',
+        '-Wpedantic',
+        '-Wswitch-enum',
+        '-Wshadow',
+        '-Wno-long-long',
+    ]
+endif
+## using MSVC headers requires c++14, if not will show an error on xstddef as: 
+## 'auto' return without trailing return type; deduced return types are a C++14 extension
+if host_machine.system() == 'windows'
+    message('[WINDOWS] using c++14 standard')
+    sqlitecpp_opts += [
+        'cpp_std=c++14',
+    ]
+    # check that we are not trying to build as dynamic library
+    if get_option('default_library') != 'shared'
+        warming('SQLiteCpp does not support shared library on Windows, the library will be built as static')
+    endif
+    
+endif
+# Options relative to SQLite and SQLiteC++ functions
+
+if get_option('SQLITE_ENABLE_COLUMN_METADATA')
+    sqlitecpp_args += [
+        '-DSQLITE_ENABLE_COLUMN_METADATA',
+    ]
+endif
+
+if get_option('SQLITE_ENABLE_ASSERT_HANDLER')
+    sqlitecpp_args += [
+        '-DSQLITE_ENABLE_ASSERT_HANDLER',
+    ]
+endif
+
+if get_option('SQLITE_HAS_CODEC')
+    sqlitecpp_args += [
+        'SQLITE_HAS_CODEC',
+    ]
+endif
+
+if get_option('SQLITE_USE_LEGACY_STRUCT')
+    sqlitecpp_args += [
+        '-DSQLITE_USE_LEGACY_STRUCT',
+    ] 
+endif
+
+## C++17 disable the support for std::filesystem (by default off)
+if get_option('SQLITECPP_DISABLE_STD_FILESYSTEM')
+    sqlitecpp_cxx_flags += ['-DSQLITECPP_DISABLE_STD_FILESYSTEM']
+endif
+
+## stack protection hardening
+if get_option('SQLITECPP_USE_STACK_PROTECTION')
+    ## if is on MinGW-W64 give a warning that is not supported
+    if mingw_64_env
+        warning('SQLiteCpp does not support stack protection on MinGW-W64')
+        warming('this could lead to a crash on the application')
+        warming('you can disable this warning by setting SQLITECPP_USE_STACK_PROTECTOR to false')
+    else
+        sqlitecpp_args += ['-fstack-protector']
+    endif
+endif
+
+## enable ommit load extension
+if get_option('SQLITE_OMIT_LOAD_EXTENSION')
+    sqlitecpp_args += ['-DSQLITE_OMIT_LOAD_EXTENSION']
+## check if running on OSX
+elif host_machine.system() == 'darwin' and sqlite3_dep.found()
+   sqlite3_load_extension_support = cxx.links(
+        '''
+            #include <sqlite3.h>
+            int main() {
+                sqlite3_enable_load_extension(0, 0);
+                return 0;
+            }
+        ''', 
+        name: 'sqlite3_load_extension',
+        dependencies: [sqlite3_dep])
+    if not sqlite3_load_extension_support
+        warming('Detected bundled SQLite3 in OSX, but it does not support load extension')
+        warming('SQLiteCpp will be built without load extension support')
+        warming('You can disable this warning by setting SQLITE_OMIT_LOAD_EXTENSION to false')
+        sqlitecpp_args += ['-DSQLITE_OMIT_LOAD_EXTENSION']
+    endif
+endif
+
+
+
+if unix_like
+    sqlitecpp_args += [
+        '-DfPIC',
```

Comment:
this is already default.

Reply:
> I thought there was an upstream meson.build file, though?

yes, however 3.2.1 was already launched today with a fix for `std::filesystem` so this time those patches should be made on this release and the next one removed, also it would be good to add a CI on that repo for meson as testing all the environments manually is hard for me

Reply:
w.r.t. `-DfPIC`, this isn't the default anywhere -- largely because it seems to have been a typo. This defines a macro called `#define fPIC`, rather than passing the `-fPIC` option, which is an `-f` (Options for Code Generation Conventions) grouped option enabling PIC code generation.

Admittedly, `-fPIC` is indeed part of Meson, and you even get a warning if you try to pass it manually.

Reply:
solved